api/src/test/java/com/epam/pipeline/controller/cluster/ClusterControllerTest.java

```diff
@@ -0,0 +1,309 @@
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("regionId", "1");
+        params.add("toolInstances", "false");
+        params.add("spot", "true");
+        Mockito.doReturn(instanceTypes).when(mockClusterApiService).getAllowedInstanceTypes(ID, true);
+
+        final MvcResult mvcResult = performRequest(get(LOAD_INSTANCE_TYPES_URL).params(params));
+
+        Mockito.verify(mockClusterApiService).getAllowedInstanceTypes(ID, true);
+        assertResponse(mvcResult, instanceTypes, NodeCreatorUtils.INSTANCE_TYPE_LIST_TYPE);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAllowedToolInstanceTypes() throws Exception {
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("regionId", "1");
+        params.add("toolInstances", "true");
+        params.add("spot", "false");
+        Mockito.doReturn(instanceTypes).when(mockClusterApiService).getAllowedToolInstanceTypes(ID, false);
+
+        final MvcResult mvcResult = performRequest(get(LOAD_INSTANCE_TYPES_URL).params(params));
+
+        Mockito.verify(mockClusterApiService).getAllowedToolInstanceTypes(ID, false);
+        assertResponse(mvcResult, instanceTypes, NodeCreatorUtils.INSTANCE_TYPE_LIST_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadAllowedInstanceAndPriceTypesForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(LOAD_ALLOWED_INSTANCE_TYPES_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAllowedInstanceAndPriceTypes() throws Exception {
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("toolId", "1");
+        params.add("regionId", "1");
+        params.add("spot", "false");
+        final AllowedInstanceAndPriceTypes allowedInstanceAndPriceTypes =
+                NodeCreatorUtils.getDefaultAllowedInstanceAndPriceTypes();
+        Mockito.doReturn(allowedInstanceAndPriceTypes).when(mockClusterApiService)
+                .getAllowedInstanceAndPriceTypes(ID, ID, false);
+
+        final MvcResult mvcResult = performRequest(get(LOAD_ALLOWED_INSTANCE_TYPES_URL).params(params));
+
+        Mockito.verify(mockClusterApiService).getAllowedInstanceAndPriceTypes(ID, ID, false);
+        assertResponse(mvcResult, allowedInstanceAndPriceTypes, NodeCreatorUtils.ALLOWED_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailGetNodeUsageStatisticsForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(String.format(NODE_USAGE_URL, NAME)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldGetNodeUsageStatistics() throws Exception {
+        final List<MonitoringStats> monitoringStats = Collections.singletonList(new MonitoringStats());
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("from", FROM_STRING.replace('T', ' '));
```

Comment:
It would be nice to avoid such date time conversions. 

We can explicitly specify request formatter and use it here to build inputs.
```java
class ClusterControllerTest {

    ...

    private final DateTimeFormatter REQUEST_FORMATTER = DateTimeFormatter.ofPattern("iso pattern without T");
    private final DateTimeFormatter DEFAULT_FORMATTER = DateTimeFormatter.ISO_LOCAL_DATE_TIME;
    private final LocalDateTime from = LocalDateTime.parse("...", DEFAULT_FORMATTER);
    private final LocalDateTime to = LocalDateTime.parse("...", DEFAULT_FORMATTER);

    @Test
    @WithMockUser
    public void shouldGetNodeUsageStatistics() throws Exception {
        final List<MonitoringStats> monitoringStats = Collections.singletonList(new MonitoringStats());
        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
        params.add("from", from.format(REQUEST_FORMATTER));

    ...
```